REGRESSION(253327@main): [ iOS ] imported/w3c/web-platform-tests/css/css-pseudo/target-text-008.html is an almost constant failure

rr-codes opened the pull request:
#### 82e61f0babb2aba02d4b377262d780f556f4cbd7
<pre>
REGRESSION(253327@main): [ iOS ] imported/w3c/web-platform-tests/css/css-pseudo/target-text-008.html is an almost constant failure
<a href="https://bugs.webkit.org/show_bug.cgi?id=243962">https://bugs.webkit.org/show_bug.cgi?id=243962</a>
rdar://98693973

Reviewed by Aditya Keerthi and Megan Gardner.

This test is failing because the snapshot occasionally captures the text indicator.
We work around this in our own tests with the `scrollToTextFragmentIndicatorEnabled()`
setting. This is not easily an option for WPT, since WPT tests the standard
configuration.

This PR works around this by disabling the text indicator if `isControlledByAutomation`
is true, which will fix the test when it is run with the WPT infrastructure.
WebKitTestRunner will also need some of its behavior to change so that the preference
is disabled when the test is run with our own infrastructure.

* LayoutTests/http/tests/scroll-to-text-fragment/no-iframe-match-expected.html:
* LayoutTests/http/tests/scroll-to-text-fragment/no-iframe-match.html:
* LayoutTests/http/tests/scroll-to-text-fragment/start-text-emoji.html:
* LayoutTests/http/tests/scroll-to-text-fragment/start-text-quote.html:
* LayoutTests/http/tests/scroll-to-text-fragment/start-text-sentence.html:
* LayoutTests/http/tests/scroll-to-text-fragment/start-text-word.html:
* LayoutTests/http/tests/scroll-to-text-fragment/word-display-none.html:
* Source/WebCore/page/FrameView.cpp:
(WebCore::FrameView::scrollToFragment):
* Tools/DumpRenderTree/TestOptions.cpp:
(WTR::TestOptions::defaults):

Canonical link: <a href="https://commits.webkit.org/256615@main">https://commits.webkit.org/256615@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/7864baf9a757b251e63a1cbb840f2bcda4148795

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/6/builds/96225 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/77/builds/5480 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/43/builds/29283 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/8/builds/105772 "Built successfully") | [✅ 🛠 🧪 win](https://ews-build.webkit.org/#/builders/10/builds/166110 "Built successfully and passed tests") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/11/builds/100203 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/76/builds/5604 "Built successfully") | [✅ 🛠 mac-debug](https://ews-build.webkit.org/#/builders/71/builds/34244 "Built successfully") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/36/builds/88609 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/12/builds/102524 "Built successfully") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/19/builds/101895 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/78/builds/4157 "Passed tests") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/61/builds/82831 "Built successfully") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/35/builds/31178 "Passed tests") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/9/builds/85998 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/3/builds/87915 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/34/builds/74005 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/66/builds/39966 "Built successfully") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/73/builds/19433 "Passed tests") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/67/builds/37640 "Built successfully") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/70/builds/20792 "Passed tests") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/74/builds/4587 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/80/builds/62 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/60/builds/43391 "Passed tests") | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/79/builds/63 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/62/builds/40059 "Passed tests") | | 
<!--EWS-Status-Bubble-End-->

megangardner commented on LayoutTests/http/tests/scroll-to-text-fragment/start-text-emoji.html:
```diff
@@ -1,4 +1,4 @@
-<!DOCTYPE html><!-- webkit-test-runner [ ScrollToTextFragmentIndicatorEnabled=false ] -->
+<!DOCTYPE html><!-- webkit-test-runner [ ScrollToTextFragmentIndicatorEnabled=true ] -->
```
Why are you setting this flag to true? I thought the point was to disable the flash in more places, not enable it sometimes?

megangardner reviewed (commented):
As long as the tests aren't being weird anymore, looks great!

hortont424 commented on Tools/DumpRenderTree/TestOptions.cpp:
```diff
@@ -93,6 +93,7 @@ const TestFeatures& TestOptions::defaults()
             { "RequiresUserGestureForAudioPlayback", false },
             { "RequiresUserGestureForMediaPlayback", false },
             { "RequiresUserGestureForVideoPlayback", false },
+            { "ScrollToTextFragmentIndicatorEnabled", false },
```
What about WKTR?

rr-codes replied:
@hortont424 do you know where I would disable it? I still haven't figured out where in WKTR to do so

rr-codes committed:
REGRESSION(253327@main): [ iOS ] imported/w3c/web-platform-tests/css/css-pseudo/target-text-008.html is an almost constant failure
https://bugs.webkit.org/show_bug.cgi?id=243962
rdar://98693973

Reviewed by Aditya Keerthi and Megan Gardner.

This test is failing because the snapshot occasionally captures the text indicator.
We work around this in our own tests with the `scrollToTextFragmentIndicatorEnabled()`
setting. This is not easily an option for WPT, since WPT tests the standard
configuration.

This PR works around this by disabling the text indicator if `isControlledByAutomation`
is true, which will fix the test when it is run with the WPT infrastructure.
WebKitTestRunner will also need some of its behavior to change so that the preference
is disabled when the test is run with our own infrastructure.

* LayoutTests/http/tests/scroll-to-text-fragment/no-iframe-match-expected.html:
* LayoutTests/http/tests/scroll-to-text-fragment/no-iframe-match.html:
* LayoutTests/http/tests/scroll-to-text-fragment/start-text-emoji.html:
* LayoutTests/http/tests/scroll-to-text-fragment/start-text-quote.html:
* LayoutTests/http/tests/scroll-to-text-fragment/start-text-sentence.html:
* LayoutTests/http/tests/scroll-to-text-fragment/start-text-word.html:
* LayoutTests/http/tests/scroll-to-text-fragment/word-display-none.html:
* Source/WebCore/page/FrameView.cpp:
(WebCore::FrameView::scrollToFragment):
* Tools/DumpRenderTree/TestOptions.cpp:
(WTR::TestOptions::defaults):

Canonical link: https://commits.webkit.org/256615@main
